pytorch_lightning/trainer/connectors/optimizer_connector.py

```diff
@@ -32,7 +32,7 @@ def update_learning_rates(self, interval: str, monitor_metrics=None):
             interval: either 'epoch' or 'step'.
             monitor_metrics: dict of possible values to monitor
         """
-        if not self.trainer.lr_schedulers:
+        if not self.trainer.lr_schedulers or not self.trainer.train_loop.automatic_optimization:
```

Comment:
Let's update the doc with this change + add a warning if we detect the scheduler was on step or epoch, but never triggered.

Reply:
I updated the docs and added a warning when scheduler keys (e.g. "interval", "frequency", "monitor"), which are invalid in manual optimization, are provided. Would you mind having a look at the changes?

Reply:
Looks good. Could we add in the doc how to use the scheduler for interval=`epoch`.

```
 def training_step(self, batch, batch_idx)

     # do optimization

      if self.trainer.is_last_batch: # is_last_batch doesn't exist for the trainer, could be added.
             self.lr_schedulers[0].step()
```

Reply:
@tchaton Addressed! Could you have a look again?